### PR TITLE
[iOS] Add toggle play pause

### PR DESCRIPTION
**Pull request by anderslemke**
This solves a few issues with lined EarPods and AirPods.

Especially that both EarPods and AirPods now can toggle play/pause. (Which they couldn't before)

### CODE REVIEW

File: ios/RNTrackPlayer/RNTrackPlayer.swift [hunk 1]
```diff
@@ -191,6 +193,11 @@ public class RNTrackPlayer: RCTEventEmitter, AudioPlayerDelegate {
             self?.sendEvent(withName: "remote-play", body: nil)
             return MPRemoteCommandHandlerStatus.success
         }
+
+        player.remoteCommandController.handleTogglePlayPauseCommand = { [weak self] _ in
```
Comment (dcvz) [inline ios/RNTrackPlayer/RNTrackPlayer.swift]: @anderslemke search for: `player.remoteCommandController.handleTogglePlayPauseCommand` -> this method already exists.. does it not work as expected?

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java [hunk 2]
```diff
@@ -113,6 +113,7 @@ private void waitForConnection(Runnable r) {
 
         // Capabilities
         constants.put("CAPABILITY_PLAY", PlaybackStateCompat.ACTION_PLAY);
+        constants.put("CAPABILITY_TOGGLE_PLAY_PAUSE", PlaybackStateCompat.ACTION_PLAY);
```
Comment (Guichaguri) [inline android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: There is also an [ACTION_PLAY_PAUSE](https://developer.android.com/reference/android/support/v4/media/session/PlaybackStateCompat.html#ACTION_PLAY_PAUSE) in Android

File: ios/RNTrackPlayer/RNTrackPlayer.swift (hunk 1 above)
Comment (appinteractive) [reply ios/RNTrackPlayer/RNTrackPlayer.swift]: Yeah it seems that it exists but wasn't used until now, corret?

**Comment by appinteractive** [comment]
Any progress @anderslemke?

**Comment by Guichaguri** [comment]
I think we should automatically set this capability based on whether `CAPABILITY_PLAY` and `CAPABILITY_PAUSE` are both set, we can also trigger `remote-play` or `remote-pause` based on the player state instead of adding a new event.

File: ios/RNTrackPlayer/RNTrackPlayer.swift (hunk 1 above)
Comment (anderslemke) [reply ios/RNTrackPlayer/RNTrackPlayer.swift]: Yes. That already exists. I'll remove this.

I don't think it was used before though, as @appinteractive points out.

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java (hunk 2 above)
Comment (anderslemke) [reply android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: OK. I'll update.

**Comment by anderslemke** [comment]
We are using this in production at Zetland now.

To answer your questions @Guichaguri :
1) I can't really tell if automagically adding the toggle capability based on play/pause is a good idea. I can't think if a situation where you wouldn't want that, but on the other hand, I don't understand why Apple would make it a separate capability in the first place.

2) The code that emitted a new event in now removed, so we use `remote-pause` and `remote-play` now.

**Comment by appinteractive** [comment]
Thanks @anderslemke for your work! 👌
@Guichaguri mergeable?

**Comment by Guichaguri** [comment]
@anderslemke I'm still not sure if we should expose those kind of API decisions to our API, but I'll leave that to @dcvz to decide.

**Comment by dcvz** [comment]
what are we doing on Android @Guichaguri ? My intuition would be to just enable `playpause` when `play` and `pause` are enabled. That's what I tried to do the first time around.

**Comment by Guichaguri** [comment]
Android does have a PLAY_PAUSE capability, but it doesn't have anything different over enabling PLAY and PAUSE separately. I think we should go with automatically enabling PLAY_PAUSE when PLAY and PAUSE are both enabled, and then splitting the event into `remote-play` and `remote-pause` depending on the state.

My guess is that the device uses the PLAY_PAUSE when it doesn't know (or is not sure about) the state of the player, like a smartwatch that doesn't have a stable connection to the phone due to battery saving reasons. And that's probably the same thing that happens with AirPods/EarPods, they may not know the current player state, so PLAY_PAUSE is triggered instead.

**Comment by mitchdowney** [comment]
This feature would be greatly appreciated :) the ability to toggle play/pause with the headphone control is very convenient.

**Comment by t-moster** [comment]
> This feature would be greatly appreciated :) the ability to toggle play/pause with the headphone control is very convenient.

Same here, we encountered the exact same problem and would love to have the feature.

**Comment by rcerrejon** [comment]
We need that feature too, would be certainly great if someone make a the review and merge the changes. I wouldn't  like to Fork this module just for that feature.

**Comment by curiousdustin** [comment]
This would be a great addition :) 

@Guichaguri , If I understand correctly the only thing left here is to add the logic to automatically enable the CAPABILITY_TOGGLE_PLAY_PAUSE  capability if both CAPABILITY_PLAY and CAPABILITY_PAUSE are provided?

**Comment by Guichaguri** [comment]
@curiousdustin Yes, remove the `CAPABILITY_TOGGLE_PLAY_PAUSE`, automatically set it when both `CAPABILITY_PLAY` and `CAPABILITY_PAUSE` are set, add logic to trigger the `remote-play` or `remote-pause` events when the play-pause button is pressed. No need for breaking changes or new API

**Comment by puckey** [comment]
I made a PR here that automatically deduces togglePlayPause from play and pause capabilities: https://github.com/react-native-kit/react-native-track-player/pull/957

**Comment by puckey** [comment]
Merged https://github.com/react-native-kit/react-native-track-player/pull/957

**Comment by biancamagalhaes** [comment]
@anderslemke i had the same problem with ios and you helped me a lot. Thanks!

**Pull request closed without merging**